pygeoapi/api.py

```diff
@@ -2102,26 +2160,93 @@ def get_collection_item(self, request: Union[APIRequest, Any],
+
+        crs_transform_wkt = None
+        if provider_type == 'feature':
+            # crs query parameter is only available for OGC API - Features
+            # right now, not for OGC API - Records.
+            LOGGER.debug('Processing crs parameter')
+            # List of CRSs supported by the server for the collection.
+            supported_crs_list = provider_def.get('crs', list())
+            for uri in DEFAULT_CRS:
+                if uri not in supported_crs_list:
+                    supported_crs_list.append(uri)
+            storage_crs_uri = provider_def.get('storage_crs')
+            crs_uri_out = request.params.get('crs')
+            if not crs_uri_out:
+                LOGGER.debug('crs parameter unspecified')
+                headers['Content-Crs'] = f'<{storage_crs_uri}>'
+            else:
+                try:
+                    crs_out = get_crs_from_uri(crs_uri_out)
+                    # Check that the crs specified by the query parameter is
+                    # supported.
+                    if crs_uri_out not in supported_crs_list:
+                        raise ValueError
+                except CRSError as err:
+                    msg = str(err)
+                    return self.get_exception(
+                        HTTPStatus.BAD_REQUEST, headers, request.format,
+                        'InvalidParameterValue', msg)
+                except ValueError:
+                    msg = (
+                        f'CRS {crs_uri_out!r} not supported for this '
+                        'collection. List of supported CRSs: '
+                        f'{", ".join(supported_crs_list)}.'
+                    )
+                    return self.get_exception(
+                        HTTPStatus.BAD_REQUEST, headers, request.format,
+                        'InvalidParameterValue', msg)
+                headers['Content-Crs'] = f'<{crs_uri_out}>'
+                # Check if the coordinates of requested features need to be
+                # transformed when served to the end-users.
+                storage_crs = get_crs_from_uri(storage_crs_uri)
+                if str(storage_crs) != str(crs_out):
+                    LOGGER.debug(
+                        f'CRS transformation: {storage_crs} -> {crs_out}'
+                    )
+                    crs_transform_wkt = CrsTransformWkt(
+                        source_crs_wkt=storage_crs.to_wkt(),
+                        target_crs_wkt=crs_out.to_wkt(),
+                    )
+                else:
+                    LOGGER.debug('No CRS transformation')
 
         # Get provider language (if any)
         prv_locale = l10n.get_plugin_locale(provider_def, request.raw_locale)
 
         try:
             LOGGER.debug(f'Fetching id {identifier}')
-            content = p.get(identifier, language=prv_locale)
+            content = p.get(
+                identifier,
+                language=prv_locale,
+                crs_transform_wkt=crs_transform_wkt,
```

Comment:
may leave out ',' .

Reply:
I have been using *Black* for the formatting of previous projects, and got used to the trailing comma. [Black's code style](https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html) points out that *"Such formatting produces smaller diffs; when you add or remove an element, it’s always just one line."* But I can remove it if it makes the code style more consistent with pygeoapi's code style.

Reply:
Ok!